docs/index.md

```diff
@@ -15,3 +15,28 @@ in the future, we might just index the metric names and then have a separate str
 
 
 prefill at startup
+
+
+## Developers' guide to index plugin writing
+
+### required query modes
+An index plugin needs to support:
+
+* lookup (1) by id (used by graphite-metrictank. may be deprecated long term)
+* lookup (1) by orgid (2) + target spec, where target spec is:
+  - a graphite key
+  - a graphite pattern that has wildcards (`*`), one of multiple options `{foo,bar}`, character lists `[abc]` and ranges `[a-z0-9]`.
+  - in the future we will want to extend these with tag constraints (e.g. must have given key, key must have given value, or value for key must match a pattern similar to above pattern)
```

Comment:
> tags are flat strings, they are not key:value pairs. If we want to have key:value pairs

you're talking about the implementation we use where we allow both `value` tags and `key:value` tags. I was talking about them more conceptually, but with our implementation that means we'll do searches like (using regex syntax here for example) `key:.*` `http_method:2..` etc.